doc/acl/Dynamic-ACL-Design.md

```diff
@@ -0,0 +1,198 @@
+
+;field          = value
+is_dynamic      = true/false  ; Boolean; Optional. Indicate the rule is a time-based ACL rule.
+creation_time   = Integer     ; timestamp when the rule is created or refreshed
+expiration_time = Integer     ; timestamp when the rule is expired
+ttl             = Integer     ; the ttl, in second
+```
+A sample config for ACL rule
+```json
+{
+    "DYNAMIC_ACL_RULE_TABLE|DATAACL|DYNAMIC_RULE_1":{
+        "DST_IP":"192.168.0.3/32",
+        "ETHER_TYPE":"2048",
+        "PACKET_ACTION":"FORWARD",
+        "PRIORITY":"9999",
+        "SRC_IP":"192.168.0.2/32",
+        "creation_time": "1662432143",
+        "expiration_time": "1662432153",
+        "is_dynamic": "True",
+        "ttl": "10"
+    }
+}
+```
+A sample json config for ACL rule
+```json
+{
+    "acl":{
+        "acl-sets":{
+            "acl-set":{
+                "DATAACL":{
+                    "acl-entries":{
+                        "acl-entry":{
+                            "1":{
+                                "actions":{
+                                    "config":{
+                                        "forwarding-action":"ACCEPT"
+                                    }
+                                },
+                                "config":{
+                                    "sequence-id":1
+                                },
+                                "ip":{
+                                    "config":{
+                                        "source-ip-address":"192.168.0.2/32",
+                                        "destination-ip-address":"192.168.0.3/32"
+                                    }
+                                },
+                                "dynamic-acl":{
+                                    "config":{
+                                        "ttl":"10"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+The YANG of ACL_RULE is required to be updated to accept new field `is_dynamic`, `ttl`, `creation_time` and `expiration_time`
+
+Orchagent (actually `aclorch`) won't consume the value of the new flags. So no change is required to `orchagent`.
+#### APP DB
+Existing table `ACL_RULE_TABLE` is used to activate an effective ACL rule. The ACL rule format is the same as in CONFIG DB.
+
+A sample
+```json
+{
+    "APP_ACL_RULE|DATAACL|DYNAMIC_RULE_1":{
+        "DST_IP":"192.168.0.3/32",
+        "ETHER_TYPE":"2048",
+        "PACKET_ACTION":"FORWARD",
+        "PRIORITY":"9999",
+        "SRC_IP":"192.168.0.2/32",
+    }
+}
+```
+#### acl-loader
+
+Update `acl-loader` script to parse new field `ttl`. The entry will be created in `DYNAMIC_ACL_RULE_TABLE` in `CONFIG DB` if `ttl` is present for certain an ACL rule in `acl.json`. Please find more details in workflow diagram.  
+
+#### dynamic_acl_mgrd
```

Comment:
Let's use `time_range_acl_mgrd` instead.

In actual, we can use `orchagent` to do this. We need to discuss pure `orchagent` based solution here

Reply:
Accepted and thanks!